jerry-core/vm/vm.c

```diff
@@ -620,11 +619,26 @@ vm_init_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
                                                      literal_start_p[value_index]);
           }
 
-          /* TODO: check the return value */
-          ecma_op_put_value_lex_env_base (ref_base_lex_env_p,
-                                          name_p,
-                                          is_strict,
-                                          lit_value);
+          if ((ecma_get_lex_env_type (frame_ctx_p->lex_env_p) == ECMA_LEXICAL_ENVIRONMENT_DECLARATIVE)
```

Comment:
I borrowed the check code from here:
https://github.com/Samsung/jerryscript/blob/master/jerry-core/parser/js/js-parser.c#L871